### DISCUSSION TITLE
Improvements to Skeletal Animation and Polygon2D

### DISCUSSION THREAD

**Discussion by FelipeRamones** [godotengine/godot-proposals#6448] [2D]
Hi everyone ^^

I'm starting this discussion just because i think not much love is being given to the 2D Skeletal animation feature in a while, i might be wrong but i feel like next to nothing has changed. The workflow is pretty rough and the manual setting of meshes and the inability to upscale proportionally the Polygons after scaling the .svg file in the import settings kinda adds a lot of time to it. Also the hability to skin characters only works if basic stuff is changed like colors and so, would be interesting to see a more robust skinning system. Been messing recently with this feature and even tho it's there feels pretty time consuming and unintuitive to work with. I would suggest some improvements to the IK features in the Skeleton2D node too, the way it is right now it's not very intuitive to set up and when set, not very good to use since it fails to detect a direction in which the joints should bend, making stuff like ewbows that bend in the worng direction happen.
I wish i had the background and time to try to contribute myself to the engine with those features but this discussion is for now the best i can do to try and bring awareness to those things a bit, i know third parties have runtimes for Godot and have way more features but they happen to be in the most cases very expensive. Would be cool to have a rag doll feature built in for this too but that goes into another topic i guess lol. 

What do you guys think about it? Would love to see some stuff being added or changed in how it works.

**Comment by SlugFiller** [comment]
I was actually able to implement this myself with GDScript. You just need a node that performs the IK calculations on _process, and make sure it's after the Skeleton2D and AnimationPlayer, so it processes later.

The SkeletonModification2D API is all sorts of broken, and will probably be deprecated and removed in the future. Doing things with additional nodes generally works much better. Although adding a way to detect if a skeleton has changed, to save on doing unnecessary recalculations, would be helpful. But, in practice, if you have an animated character, they would be changing every frame anyway.
